packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/InternalMutableRealm.kt

```diff
@@ -66,4 +67,22 @@ internal interface InternalMutableRealm : MutableRealm {
     override fun delete(deleteable: Deleteable) {
         deleteable.asInternalDeleteable().delete()
     }
+
+    override fun delete(schemaClass: KClass<out BaseRealmObject>) {
+        try {
+            delete(query(schemaClass).find())
+        } catch (err: IllegalStateException) {
+            if (err.message?.contains("not part of this configuration schema") == true) {
+                throw IllegalArgumentException()
```

Comment:
I would pass on the message here, so `throw IllegalArgumentException(err.message)`